lib/exq/middlewares/stats.ex

```diff
@@ -0,0 +1,46 @@
+defmodule Exq.Middlewares.Stats do
+  @behaviour Exq.Middlewares.Behaviour
+  require Logger
+  alias Exq.Stats.Server, as: Stats
+  import Exq.Middlewares.Pipeline
+
+  def before_work(pipeline) do
+    {:ok, info} = Stats.add_process(pipeline.assigns.stats, pipeline.assigns.namespace,
+      pipeline.worker_pid, pipeline.assigns.host, pipeline.assigns.job_json)
+
+    pipeline = assign(pipeline, :process_info, info)
+    pipeline
+  end
+
+  def after_processed_work(pipeline) do
+    case is_alive?(pipeline.assigns.stats) do
```

Comment:
I think with supervisor order above we don't need the `alive?` checks so can probably remove a bunch of code from here (and methods below).